### PR TITLE
feat(rest/ui): Project vulnerability endpoint update & added new project rating in UI

### LINKED ISSUE #1024: REST/UI: Project Vulnerability enhancement
1) In UI of projects->vulnerabilities, Currently the column: 'Relevance for Projects' has currently the possible states:
Not Checked
irrelevant
Resolved
applicable
For partial automation of our vulnerability-handling-process we need an additional state such as `In Analysis`. 

2)  REST API: Projects/Vulnerabilities - please provide 'Priority', 'Relevance for Projects' and 'Actions' fields via REST API
3)  REST API: Projects/Vulnerabilities - please add 'comment' field and make it available via REST API

### PR DESCRIPTION
**Pull request by smrutis1**
[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.

 Which issue is this pull request belonging to and how is it solving it? (#1024 )
 Did you add or update any new dependencies that are required for your change? - No

Issue:  closes #1024 

### How To Test?

* Go to  Project-> Vulnerabilities, a new project rating `In Analysis` should be visible in the dropdown.
* Need to test project vulnerabilities endpoints as per the newly added documentation. 

Have you implemented any additional tests? - Yes, for REST documentation

### Checklist
Must:
- [ ] All related issues are referenced in commit messages and in PR

Signed-off-by: Smruti Prakash Sahoo <smruti.sahoo@siemens.com>

### CODE REVIEW

**Review by JaideepPalit — CHANGES_REQUESTED** [review]
Reviewed and tested the PR.
Kindly look into some minor review comments and observations.

**Observations**
- request structure in PATCH API doc is not correct
- comment is not coming in response and doc for `get-project-vulnerability`
- Non-existing external Id and release ID gives 200 ok response in PATCH response

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java [hunk 1]
```diff
@@ -14,6 +14,7 @@
 import lombok.NonNull;
 import lombok.RequiredArgsConstructor;
 import org.apache.commons.lang.StringUtils;
+import org.apache.http.HttpEntity;
```
[the remark this answers is not in the document]
Comment (smrutis1) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java]: Removed it

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/vulnerability/Sw360VulnerabilityService.java [hunk 2]
```diff
@@ -88,4 +95,52 @@ public Vulnerability getVulnerabilityWithReleaseRelationsByExternalId(String ext
...
+        String releaseId = vulnDto.getIntReleaseId();
+        if (!vulnerabilityRatings.containsKey(vulnerabilityId)) {
+            vulnerabilityRatings.put(vulnerabilityId, new HashMap<>());
+        }
+        List<VulnerabilityCheckStatus> vulnerabilityCheckStatusHistory = null;
+        if (vulnerabilityIdToStatusHistory.containsKey(vulnerabilityId)
+                && vulnerabilityIdToStatusHistory.get(vulnerabilityId).containsKey(releaseId)) {
+            vulnerabilityCheckStatusHistory = vulnerabilityIdToStatusHistory.get(vulnerabilityId).get(releaseId);
+        }
+        if (vulnerabilityCheckStatusHistory != null && vulnerabilityCheckStatusHistory.size() > 0) {
```
[the remark this answers is not in the document]
Comment (smrutis1) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/vulnerability/Sw360VulnerabilityService.java]: Used `isNotEmpty(_)` one

**Comment by smrutis1** [comment]
@JaideepPalit , Thanks for review and test. Have rectified the issues. Now if there is any invalid externalId or releaseId, it would return a `400 Bad Request` along with appropriate message.

**Comment by JaideepPalit** [comment]
**Retested the PR and Observations**
> 
> * request structure in PATCH API doc is not correct 
      - looks fine
> * comment is not coming in response and doc for `get-project-vulnerability`
      - `working fine`
> * Non-existing external Id and release ID gives 200 ok response in PATCH response
      - `working fine`

**Issue**
Get request for new Project - `working fine`
Get request is not working after doing a patch - 500 error, message - null
```
2020-10-23 09:08:04 ERROR ProcessFunction:47 - Internal error processing getVulnerabilitiesByProjectId
org.eclipse.sw360.datahandler.thrift.SW360Exception: null
    at org.eclipse.sw360.datahandler.common.SW360Assert.fail(SW360Assert.java:134) ~[datahandler-11.1.0-SNAPSHOT.jar:?]
    at org.eclipse.sw360.datahandler.common.SW360Assert.assertNotNull(SW360Assert.java:42) ~[datahandler-11.1.0-SNAPSHOT.jar:?]
    at org.eclipse.sw360.datahandler.common.SW360Assert.assertNotNull(SW360Assert.java:36) ~[datahandler-11.1.0-SNAPSHOT.jar:?]
    at org.eclipse.sw360.datahandler.db.ProjectDatabaseHandler.getProjectById(ProjectDatabaseHandler.java:256) ~[src-common-11.1.0-SNAPSHOT.jar:?]
    at org.eclipse.sw360.vulnerabilities.VulnerabilityHandler.getVulnerabilitiesByProjectId(VulnerabilityHandler.java:92) ~[src-vulnerabilities-11.1.0-SNAPSHOT.jar:?]
    at org.eclipse.sw360.datahandler.thrift.vulnerabilities.VulnerabilityService$Processor$getVulnerabilitiesByProjectId.getResult(VulnerabilityService.java:1506) ~[datahandler-11.1.0-SNAPSHOT.jar:?]
    at org.eclipse.sw360.datahandler.thrift.vulnerabilities.VulnerabilityService$Processor$getVulnerabilitiesByProjectId.getResult(VulnerabilityService.java:1486) ~[datahandler-11.1.0-SNAPSHOT.jar:?]
    at org.apache.thrift.ProcessFunction.process(ProcessFunction.java:38) [libthrift-0.13.0.jar:0.13.0]
    at org.apache.thrift.TBaseProcessor.process(TBaseProcessor.java:38) [libthrift-0.13.0.jar:0.13.0]
```

**Comment by smrutis1** [comment]
Rebased and fixed the reported issue.

**Review by JaideepPalit — APPROVED** [review]
Code looks fine to me.
Tested successfully.

**Pull request merged**